src/Bicep.Core/SemanticModel/Namespaces/SystemNamespaceSymbol.cs

```diff
@@ -78,47 +72,52 @@ public class SystemNamespaceSymbol : NamespaceSymbol
-            FunctionOverload.CreateFixed("less", LanguageConstants.Bool, LanguageConstants.String, LanguageConstants.String),
-            FunctionOverload.CreateFixed("less", LanguageConstants.Bool, LanguageConstants.Int, LanguageConstants.Int),
-
-            // TODO: Needs number type
-            FunctionOverload.CreateFixed("lessOrEquals", LanguageConstants.Bool, LanguageConstants.String, LanguageConstants.String),
-            FunctionOverload.CreateFixed("lessOrEquals", LanguageConstants.Bool, LanguageConstants.Int, LanguageConstants.Int),
-
-            // TODO: Needs number type
-            FunctionOverload.CreateFixed("greater", LanguageConstants.Bool, LanguageConstants.String, LanguageConstants.String),
-            FunctionOverload.CreateFixed("greater", LanguageConstants.Bool, LanguageConstants.Int, LanguageConstants.Int),
-
-            // TODO: Needs number type
-            FunctionOverload.CreateFixed("greaterOrEquals", LanguageConstants.Bool, LanguageConstants.String, LanguageConstants.String),
-            FunctionOverload.CreateFixed("greaterOrEquals", LanguageConstants.Bool, LanguageConstants.Int, LanguageConstants.Int),
-
-            FunctionOverload.CreateFixed("equals", LanguageConstants.Bool, LanguageConstants.Any, LanguageConstants.Any),
             FunctionOverload.CreateFixed("json", LanguageConstants.Any, LanguageConstants.String),
-            FunctionOverload.CreateFixed("not", LanguageConstants.Bool, LanguageConstants.Bool),
-            FunctionOverload.CreateWithVarArgs("and", LanguageConstants.Bool, 2, LanguageConstants.Bool),
-            FunctionOverload.CreateWithVarArgs("or", LanguageConstants.Bool, 2, LanguageConstants.Bool),
-
-            // TODO: Return type should be a type tranformation
-            FunctionOverload.CreateFixed("if", LanguageConstants.Any, LanguageConstants.Bool, LanguageConstants.Any, LanguageConstants.Any),
-
+            
             new FunctionOverload("dateTimeAdd", LanguageConstants.String, 2, 3, Enumerable.Repeat(LanguageConstants.String, 3), null),
 
             // newGuid and utcNow are only allowed in parameter default values
             new FunctionOverload("utcNow", LanguageConstants.String, 0, 1, Enumerable.Repeat(LanguageConstants.String, 1), null, FunctionFlags.ParamDefaultsOnly),
             new FunctionOverload("newGuid", LanguageConstants.String, 0, 0, Enumerable.Empty<TypeSymbol>(), null, FunctionFlags.ParamDefaultsOnly),
         }.ToImmutableArray();
 
-        public SystemNamespaceSymbol() : base("sys", SystemOverloads)
+        // TODO: Add copyIndex here when we support loops.
+        private static readonly ImmutableArray<BannedFunction> BannedFunctions = new[]
+        {
+            /*
+             * The true(), false(), and null() functions are not included in this list because
+             * we parse true, false and null as keywords in the lexer, so they can't be used as functions anyway.
+             */
+
```

Comment:
Worth adding tests for this anyway in case we ever decide to not treat these as keywords?

Reply:
Yeah good pt. I'll add some.

Reply:
Added tests